pallets/parachain-staking/src/set.rs

```diff
@@ -87,3 +89,78 @@ impl<T: Ord> From<Vec<T>> for OrderedSet<T> {
 		Self::from(v)
 	}
 }
+
+/// An ordered set backed by `BoundedVec`
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(RuntimeDebug, PartialEq, Eq, Encode, Decode, Clone, TypeInfo)]
+#[scale_info(skip_type_params(S))]
+pub struct BoundedOrderedSet<T, S: Get<u32>>(pub BoundedVec<T, S>);
```

Comment:
Could this fully replace `OrderedSet`? ...in theory there should be nothing unbounded...

Reply:
Yeah in a follow up PR I'm cleaning up all the nested unbounded structures

Reply:
https://opslayer.atlassian.net/browse/MOON-2512